modules/KIWIImage.pm

```diff
@@ -4125,9 +4125,17 @@ sub extractLinux {
     my $xml       = $this->{xml};
     my $vconf     = $xml -> getVMachineConfig();
     my $xendomain;
-    my $kernname = "vmlinuz";
-    if (-f "$imageTree/boot/vmlinux") {
-        $kernname = "vmlinux";
+    my $kernname;
+    foreach (
+        "image",   # s390x
+        "vmlinux", # ppc*
+        "Image",   # aarch64
+        "uImage",  # armv* (legacy)
+        "zImage",  # armv*
+        "vmlinuz", # x86*
+        ) {
```

Comment:
ok, the KIWILinuxRC.sh change might be redundant then, I'll remove it. 

The main issue I ran into is that extractkernel was not copying out the kernel it used into the destination directory, so steps further down the line failed on that.